untrusted/test_enclave/build.rs

```diff
@@ -0,0 +1,253 @@
+    // be directly linked in.
+    Build::new().files(files)
+        .include(format!("{}/include", sgx_library_path()))
+        .include(format!("{}/include/tlibc", sgx_library_path()))
+        .cargo_metadata(false) .shared_flag(true).compile("enclave");
+
+    let static_enclave = out_dir().join("libenclave.a");
+    let dynamic_enclave = build_dynamic_enclave_binary(static_enclave);
+    sign_enclave_binary(dynamic_enclave)
+}
+
+/// Create a dynamic version of the enclave.  This is *unsigned*.
+///
+/// See https://github.com/alexcrichton/cc-rs/issues/250 for lack of dynamic
+/// lib in cc crate
+///
+/// # Arguments
+///
+/// * `static_enclave` - The static enclave binary
+///
+/// # Returns
+/// The full path to resultant shared library file.
+fn build_dynamic_enclave_binary<P: AsRef<Path>>(static_enclave: P) -> PathBuf {
+    let mut dynamic_enclave = PathBuf::from(static_enclave.as_ref());
+    dynamic_enclave.set_extension("so");
+
+    let mut command = Command::new(ld_linker());
+    command
+        .arg("-o")
+        .arg(dynamic_enclave.to_str().expect("Invalid UTF-8 in static enclave path"))
+        .args(&["-z", "relro", "-z", "now", "-z", "noexecstack"])
+        .arg(&format!("-L{}/lib64/cve_2020_0551_load", sgx_library_path()))
+        .arg(&format!("-L{}/lib64", sgx_library_path()))
+        .arg("--no-undefined")
+        .arg("--nostdlib")
+        .arg("--start-group")
+        .args(&["--whole-archive", "-lsgx_trts_sim", "--no-whole-archive"])
+        .arg(static_enclave.as_ref().to_str().unwrap())
+        .args(&["-lsgx_tstdc", "-lsgx_tcxx", "-lsgx_tcrypto", "-lsgx_tservice_sim"])
+        .arg("--end-group")
+        .arg("-Bstatic")
+        .arg("-Bsymbolic")
+        .arg("--no-undefined")
+        .arg("-pie")
+        .arg("-eenclave_entry")
+        .arg("--export-dynamic")
+        .args(&["--defsym", "__ImageBase=0"])
+        .arg("--gc-sections")
+        .arg(&format!("--version-script={}", ENCLAVE_LINKER_SCRIPT));
+
+    let status = command.status().expect("Failed to run the linker for dynamic enclave");
+    match status.code().unwrap() {
+        0 => (),
+        _ => panic!("Failed to link the dynamic enclave")
+    }
+    dynamic_enclave
+
+}
+
+/// Sign the enclave binary
+///
+/// # Arguments
+///
+/// * `unsigned_enclave` - The unsigned enclave binary
+///
+/// # Returns
+/// The full path to signed binary file.  This binary will be signed and
+/// ready for use in `sgx_create_enclave()`.
+fn sign_enclave_binary<P: AsRef<Path>>(unsigned_enclave: P) -> PathBuf {
+    let mut signed_binary = PathBuf::from(unsigned_enclave.as_ref());
+    signed_binary.set_extension("signed.so");
+
+    let mut command = Command::new(format!("{}/bin/x64/sgx_sign", sgx_library_path()));
+    command.arg("sign") .arg("-enclave") .arg(unsigned_enclave.as_ref())
+        .arg("-config") .arg(ENCLAVE_CONFIG)
+        .arg("-key") .arg(SIGNING_KEY)
+        .arg("-out") .arg(&signed_binary);
```

Comment:
Nit: Move calls one per line
```suggestion
    command
        .arg("sign")
        .arg("-enclave")
        .arg(unsigned_enclave.as_ref())
        .arg("-config")
        .arg(ENCLAVE_CONFIG)
        .arg("-key")
        .arg(SIGNING_KEY)
        .arg("-out")
        .arg(&signed_binary);
```